drivers/hmis/app/graphql/mutations/upload_client_file.rb

```diff
@@ -0,0 +1,51 @@
+# ! This is not included in the schema, but it does define the logic for creating a new file, so leaving in place as reference for the file processor
+
+module Mutations
+  class UploadClientFile < BaseMutation
+    argument :client_id, ID, required: true
+    argument :enrollment_id, ID, required: false
+    argument :file_blob_id, ID, required: true
+    argument :file_tags, [ID], required: false
+    argument :effective_date, GraphQL::Types::ISO8601Date, required: false
+    argument :expiration_date, GraphQL::Types::ISO8601Date, required: false
+    argument :confidential, Boolean, required: false
+    argument :name, String, required: false
+
+    field :client, Types::HmisSchema::Client, null: true
+
+    def resolve(client_id:, enrollment_id: nil, file_blob_id:, file_tags: [], **input)
+      client = Hmis::Hud::Client.visible_to(current_user).find_by(id: client_id)
+      enrollment = Hmis::Hud::Enrollment.viewable_by(current_user).find_by(id: enrollment_id)
+
+      errors = HmisErrors::Errors.new
+      errors.add :client_id, :not_found unless client.present?
+      errors.add :enrollment_id, :not_found if enrollment_id.present? && enrollment.nil?
+      errors.add :client_id, :not_allowed if client.present? && !current_user.can_edit_clients_for?(client)
+      errors.add :enrollment_id, :not_allowed if enrollment.present? && !current_user.can_edit_enrollments_for?(enrollment)
```

Comment:
Somewhere in here I think it should be checking that the user `can_manage_client_files`

Reply:
This mutation is dead code (it's not included in `HmisSchema::MutationType`) and it will be removed and replaced with a FileProcessor, but we split that into [#184713017](https://www.pivotaltracker.com/story/show/184713017). I think Zach is just leaving the code here as reference for the processing work in case he needs to reference any of this logic. (See comment at the top). Fine with me to leave this file here.

Reply:
^^ Correct